homeassistant/components/ozw/cover.py

```diff
@@ -73,11 +84,22 @@ async def async_set_cover_position(self, **kwargs):
 
     async def async_open_cover(self, **kwargs):
         """Open the cover."""
-        self.values.primary.send_value(99)
+        self.values.open.send_value(PRESS_BUTTON)
```

Comment:
How does this impact covers that used the primary value before? Do all devices support the button presses?

Reply:
When checking to `zwave/cover.py` component, the button press is the behavior used. Sending a extremum value was a workaround while button values were not supported through openzwave / mqtt.
What could be wrong in the current implementation is the lack of support for inverted buttons or cover position (it is supported by `zwave/cover.py`), if we find a device where buttons are inverted but the position is not.

Reply:
Someone should verify that the behavior is the same between OpenZWave 1.4 and 1.6.

Reply:
> How does this impact covers that used the primary value before? Do all devices support the button presses?

All window coverings (not garage doors) are required to support the Multilevel Switch Command Class. The `open` and `close` values are [mapped by HA](https://github.com/home-assistant/core/blob/3de0610909e97f92a09a9efae117e0fabfeb0067/homeassistant/components/ozw/discovery.py#L183) (for both ozw and zwave) to the button values that OZW creates called "Bright" and "Dim", which then correspond to the  Z-Wave Multilevel Switch commands Start Level Change and Stop Level Change. Therefore all window coverings support this button API, since the Start/Stop Level Change commands have existed since V1. 

The primary value would be the "Level" value itself, which sets the "position" (brightness).

> Someone should verify that the behavior is the same between OpenZWave 1.4 and 1.6.

The implementations of the button values are nearly identical ([1.4 hass](https://github.com/home-assistant/open-zwave/blob/hass/cpp/src/command_classes/SwitchMultilevel.cpp#L290) vs [1.6](https://github.com/OpenZWave/open-zwave/blob/4fe9af833a6d1fc8949918bf117e131378b7d059/cpp/src/command_classes/SwitchMultilevel.cpp#L250)). 

A button press with the `open` value triggers a Start Level Command with the direction "up". A button press with the `close` value triggers a Start Level Command with the direction "down". A button release will issue a Stop Level Command. It doesn't matter which button you use for the "release" action, either one triggers the same result, which is why the zwave integration seems to have picked the `open` value for stop (or close value if inverted).

The implementation of `SwitchMultilevel::StartLevelChange` (which takes the up or down direction) looks about the same to me.

As for ozwdaemon, it [uses](https://github.com/OpenZWave/qt-openzwave/blob/master/docs/MQTT.md#setvalue) true for a button press and false for a release, and [calls](https://github.com/OpenZWave/qt-openzwave/blob/7ebd43e246e97851ad20f2b1bb9c62bfd6a4a6ef/qt-openzwave/source/qtozwmanager_p.cpp#L2206) the same pressButton and releaseButton API that the zwave integration uses.

Reply:
@kpine thank you! Great insight!